internal/venice-client-common/src/main/java/com/linkedin/venice/D2/D2ClientUtils.java

```diff
@@ -62,6 +62,9 @@ public static void shutdownClient(D2Client client) {
   }
 
   public static void shutdownClient(D2Client client, long timeoutInMs) {
+    if (client == null) {
```

Comment:
why do we call shutdown on a null client?

Reply:
It happens in cases where the setup of a test has multiple steps, some early step fails and a latter step of spinning up a d2 client gets short-circuited, and then the cleanup function of the test still runs, but passes null into this function.